File: 9-palindrome-number/palindrome-number.py
class Solution:
    def reverseNum(self, num: int) -> int:
        revNum = 0
        while num > 0:
           revNum = (revNum * 10) + (num % 10)
           num = num // 10

        return revNum

    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            if self.reverseNum(x) - x == 0:
                return True
            else:
                return False

 

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        std::string num = std::to_string(x);
        int left = 0, right = num.length() - 1;
        while (left < right) {
            std::cout << left << num[left] << right << num[right];
            if (num[left] != num[right]) return false;
            left += 1;
            right -= 1;
        }
        return true;
    }
};